buf/registry/module/v1beta1/commit_service.proto

```diff
@@ -17,341 +17,135 @@ syntax = "proto3";
-  //   - If a Digest is referenced, this is interpreted to mean the latest released Commit that has this Digest.
-  //     Digests referencing unreleased Commits cannot be referenced.
-  //
-  // TODO: what is "latest released Commit" if we now have "head Commit"?
-  repeated ResourceRef resource_refs = 1 [
+message GetCommitsRequest {
+  // The Commits to request.
+  repeated CommitRef commit_refs = 1 [
     (buf.validate.field).repeated.min_items = 1,
     (buf.validate.field).repeated.max_items = 250
   ];
+  // The DigestType to use for Digests returned on Commits.
+  //
+  // If this DigestType is not available, an error is returned.
+  // Note that certain DigestTypes may be deprecated over time.
+  //
+  // If not set, the latest DigestType is used, currently B5.
+  DigestType digest_type = 2;
 }
 
-message ResolveCommitsResponse {
-  // The found Commits in the same order as requested.
+message GetCommitsResponse {
+  // The retrieved Commits in the same order as requested.
   repeated Commit commits = 1 [(buf.validate.field).repeated.min_items = 1];
 }
 
-message ListCommitHistoryRequest {
+message ListCommitsRequest {
+  // The list order.
+  enum Order {
+    ORDER_UNSPECIFIED = 0;
+    // Order by create_time oldest to newest.
+    ORDER_CREATE_TIME_ASC = 1;
+    // Order by create_time newest to oldest.
+    ORDER_CREATE_TIME_DESC = 2;
+  }
   // The maximum number of items to return.
   //
   // The default value is 10.
   uint32 page_size = 1 [(buf.validate.field).uint32.lte = 250];
   // The page to start from.
   //
-  // If empty, the first page is returned.
+  // If empty, the first page is returned,
   string page_token = 2 [(buf.validate.field).string.max_len = 4096];
-  // The reference to get the history for.
-  //
-  // See the documentation on ResourceRef for resource resolution details.
+  // The reference to list Commits for.
   //
-  // Once the resource is resolved, history is started from the following Commit:
-  //   - If a Module is referenced, history is started at the latest released Commit on the Module.
-  //   - If a Commit is referenced, history is started at this Commit.
-  //   - If a Tag is referenced, history is started at the Commit associated with the Tag.
-  //   - If a VCSCommit is referenced, history is started at the Commit associated with the VCSCommit.
-  //   - Is a Branch is referenced, history is started at the head Commit on the Branch.
-  //   - If a Digest is referenced, history is started at the latest released Commit that has this Digest.
-  //     Digests referencing unreleased Commits cannot be referenced.
+  // See the documentation on Ref for resource resolution details.
   //
-  // TODO: what is "latest released Commit" if we now have "head Commit"?
+  // Once the resource is resolved, the following Commits are listed:
+  //   - If a Module is referenced, all Commits for the Module are returned.
```

Comment:
This should probably be scoped to the default label instead of mixing commits across all labels
```suggestion
  //   - If a Module is referenced, all Commits for the default Label of the Module are returned.
```